### PR TITLE
Eth 362 reset trusted forwarder

### PR DESCRIPTION
**Pull request by samt1803**

### CODE REVIEW

File: packages/smartcontracts/scripts/deployToLivenet/2_interactWithContracts.ts [hunk 1]
```diff
@@ -48,9 +50,10 @@ const DEPLOYMENT_OWNER_KEY = '0x4059de411f15511a85ce332e7a428f36492ab4e87c783009
 // const CHAINLINK_NODE_ADDRESS = '0x7b5F1610920d5BAf00D684929272213BaF962eFe'
 
 // addresses localsidechain
+const FORWARDER = '0xEaa9846004d98BC4aa79059cF70d528235e59a6E'
 const ORACLEADDRESS = '0xD94D41F23F1D42C51Ab61685e5617BBC858e5871'
 const ENSCACHEADDRESS = '0xE4eA76e830a659282368cA2e7E4d18C4AE52D8B3'
-const STREAMREGISTRYADDRESS = '0x6cCdd5d866ea766f6DF5965aA98DeCCD629ff222'
+const STREAMREGISTRYADDRESS = '0x382b486B81FefB1F280166f2000a53b961b9840d'
```
Comment (jtakalai) [inline packages/smartcontracts/scripts/deployToLivenet/2_interactWithContracts.ts:56]: could it help making these scripts more reusable if these would come from process.env, then maybe for UI ease of use, we could put them in `.env` file or something? Doesn't need to be changed in this ticket, but clearly if these addresses change often, it's kind of noisy to commit them?

File: packages/smartcontracts/test/StreamStorageRegistry.test.ts [hunk 2]
```diff
@@ -148,4 +201,120 @@ describe('StreamStorageRegistry', () => {
...
+        const options = {
+            data: d,
+            privateKey: ethers.utils.arrayify(signKey) as Buffer,
+            version: SignTypedDataVersion.V4,
+        }
+        const sign = signTypedData(options) // user0
+        return {req, sign}
+    }
+
+    it('positivetest metatransaction', async (): Promise<void> => {
```
Comment (jtakalai) [inline packages/smartcontracts/test/StreamStorageRegistry.test.ts:236]: 👍

**Review by jtakalai — APPROVED** [review]

File: .vscode/launch.json [hunk 3]
```diff
@@ -62,7 +62,7 @@
             "type": "node",
             "request": "launch",
             "name": "Hardhat test this file",
-            "runtimeExecutable": "${workspaceFolder}/packages/smartcontracts/node_modules/.bin/hardhat",
+            "runtimeExecutable": "${workspaceFolder}/node_modules/.bin/hardhat",
```
[the remark this answers is not in the document]
Comment (samt1803) [reply .vscode/launch.json]: done

File: packages/smartcontracts/scripts/deployToLivenet/2_interactWithContracts.ts (hunk 1 above)
Comment (samt1803) [reply packages/smartcontracts/scripts/deployToLivenet/2_interactWithContracts.ts]: reverted it, we can think of a better way to get the addresses from one script to the other

File: packages/smartcontracts/contracts/StreamRegistry/ERC2771ContextUpgradeable.sol [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    function _msgData() internal view virtual override returns (bytes calldata) {
+        if (isTrustedForwarder(msg.sender)) {
+            return msg.data[:msg.data.length - 20];
+        } else {
+            return super._msgData();
+        }
+    }
+    uint256[49] private __gap;
+
+    function _setTrustedForwarder(address trustedForwarder) internal {
```
[the remark this answers is not in the document]
Comment (samt1803) [reply packages/smartcontracts/contracts/StreamRegistry/ERC2771ContextUpgradeable.sol:53]: done

File: packages/smartcontracts/contracts/StreamStorageRegistry/StreamStorageRegistryV2.sol [hunk 5]
```diff
@@ -0,0 +1,99 @@
+/**
+ * Not yet deployed
+ */
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+pragma experimental ABIEncoderV2;
```
[the remark this answers is not in the document]
Comment (samt1803) [reply packages/smartcontracts/contracts/StreamStorageRegistry/StreamStorageRegistryV2.sol]: done

**Pull request merged**